core-processor/src/main/java/io/micronaut/inject/beans/visitor/BeanIntrospectionWriter.java

```diff
@@ -419,32 +419,95 @@ private void buildStaticInit(ClassWriter classWriter) {
     private void pushBeanPropertyReference(ClassWriter classWriter,
                                            GeneratorAdapter staticInit,
                                            BeanPropertyData beanPropertyData) {
-        staticInit.newInstance(Type.getType(AbstractInitializableBeanIntrospection.BeanPropertyRef.class));
-        staticInit.dup();
 
-        pushCreateArgument(
+        Runnable pushTypeArgument = () -> pushCreateArgument(
             annotationMetadata,
-            beanType.getClassName(),
+            classElement.getName(),
             introspectionType,
             classWriter,
             staticInit,
             beanPropertyData.name,
-            beanPropertyData.typedElement,
-            beanPropertyData.annotationMetadata,
-            beanPropertyData.typeArguments,
+            beanPropertyData.type,
             defaults,
             loadTypeMethods
         );
+
+        int typeLocal = -1;
```

Comment:
Could use Supplier.memoized

Reply:
meh probably not more readable though

Reply:
That would be more confusing; the invocation is contextual, and if a variable is not needed, we don't create it.